Adding hasSameBinaryContentAs

ngeor opened the pull request:
#### Check List:
* Fixes #1059 
* Unit tests : YES
* Javadoc with a code example (on API only) : YES

joel-costigliola commented on src/test/java/org/assertj/core/api/file/FileAssert_hasSameBinaryContentAs_Test.java:
```diff
@@ -0,0 +1,78 @@
+    File expected = createDeleteOnExitTempFileWithContent("assertJ++");
+    assertThatExceptionOfType(AssertionError.class).isThrownBy(() -> {
+      assertThat(actual).hasSameBinaryContentAs(expected);
+    }).withMessageContaining("does not have expected binary content at offset");
```
prefer `expectAssertionError` as per https://github.com/joel-costigliola/assertj-core/blob/master/CONTRIBUTING.md
add GIVEN WHEN THEN sections

joel-costigliola commented on src/test/java/org/assertj/core/api/file/FileAssert_hasSameBinaryContentAs_Test.java:
```diff
@@ -0,0 +1,78 @@
+    }).withMessageContaining("does not have expected binary content at offset");
+  }
+
+  private File createDeleteOnExitTempFileWithContent(String content) throws IOException {
```
rename to `createTempFileWithContent`

scordio commented:
Closing and reopening to trigger Travis again

ngeor commented:
> we need to deprecate `hasSameContentAs` for `hasSameTextualContentAs` to fix #1059 otherwise
> just some minor comments, thanks @ngeor!

Thanks, I  processed the minor code changes (I should've read the Contributing guidelines first).
Just to double check, shall I introduce this new method `hasSameTextualContentAs`?

joel-costigliola commented:
Yes let's add `hasSameTextualContentAs` (a copy of `hasSameContentAs`)

ngeor commented:
I added the new method and marked the old ones as deprecated. Some extra things:
- I added a `@since` tag to the new methods
- I added a `TempFileUtil` class because the `createTempFile` method was copy pasted around
- Fixed a typo in the docs `Turkisch -> Turkish`

joel-costigliola commented:
Integrated thanks @ngeor! (I have added the same assertion for path)
